WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java

```diff
@@ -177,12 +178,16 @@ public static void addMedia(Activity activity) {
     }
 
     public static void viewMyProfile(Context context) {
+        AccountHelper.getDefaultAccount().fetchAccountSettings();
+
         Intent intent = new Intent(context, MyProfileActivity.class);
         AnalyticsUtils.trackWithCurrentBlogDetails(AnalyticsTracker.Stat.OPENED_MY_PROFILE);
         slideInFromRight(context, intent);
     }
 
     public static void viewAccountSettings(Activity activity) {
+        AccountHelper.getDefaultAccount().fetchAccountSettings();
```

Comment:
The caller in this case is `MeFragment` and I don't think it makes a lot of sense to start a fetch there. I also think it defeats the whole purpose of `ActivityLauncher` which is to remove duplication and make sure the activities are launched in a consistent way across the app. If we were to add "My Profile" into the signup process, it'd mean that we'd also need to start a fetch from there.

The best way to solve this would be to let the lower layers handle the regular fetch only when it's necessary. But out of 3 different places where we can put this code right now, I think this one makes the most sense. I could be totally wrong though :)

@maxme any thoughts on this?